common/credentialFactory.go

```diff
@@ -285,3 +285,44 @@ func (f *S3ClientFactory) GetS3Client(ctx context.Context, credInfo CredentialIn
 		return s3Client, nil
 	}
 }
+
+// ====================================================================
+// GCP credential factory related methods
+// ====================================================================
+func CreateGCPClient(ctx context.Context) (*storage.Client, error) {
+	client, err := storage.NewClient(ctx)
+	return client, err
+}
+
+type GCPClientFactory struct {
+	gcpClients map[CredentialInfo]*storage.Client
+	lock       sync.RWMutex
+}
```

Comment:
Would sync.Map be preferable here?

Reply:
Yes, but not gonna impact perf. This is help per jptm, and is upper bound. I'll skip this, makes this look different from S3 code for no tangible gain.